core/include/detray/tools/line_stepper.hpp

```diff
@@ -48,23 +51,30 @@ class line_stepper final : public base_stepper<track_t> {
      * @return returning the heartbeat, indicating if the stepping is alive
      */
     template <typename navigation_state_t>
-    DETRAY_HOST_DEVICE bool step(
-        state &stepping, navigation_state_t &navigation,
-        scalar max_step_size = std::numeric_limits<scalar>::max()) {
+    DETRAY_HOST_DEVICE bool step(state &stepping,
+                                 navigation_state_t &navigation) {
 
         // Distance to next surface as fixed step size
         scalar step_size = navigation();
 
-        // Inform navigator
+        // Update navigation direction
+        const step::direction dir = step_size > 0 ? step::direction::e_forward
+                                                  : step::direction::e_backward;
+        stepping.set_direction(dir);
+
+        // Resolve how the navigator should perform an update
+        // nav_policy(stepping, navigation);
         // Not a severe change to track state expected
-        if (step_size < max_step_size) {
+        if (step_size <
+            stepping.constraints().template size<>(stepping.direction())) {
```

Comment:
Does this work for backward direction?

Reply:
Nope, forgot the ```std::abs```. Backward navigation should not happen with a line stepper, but this needs to be fixed of course